pkg/extension/opampextension/opamp_agent_test.go

```diff
@@ -213,3 +214,53 @@ func TestReload(t *testing.T) {
 	assert.NoError(t, o.Start(ctx, componenttest.NewNopHost()))
 	assert.NoError(t, o.Reload(ctx))
 }
+
+// To be removed when the OpAMP backend correctly advertises its URL.
+// This test is badly written, it reaches into unexported fields to test
+// their contents, but given that it will be removed in a few months, that
+// shouldn't matter too much from a big picture perspective.
+func TestHackSetEndpoint(t *testing.T) {
+	tests := []struct {
+		name         string
+		url          string
+		wantEndpoint string
+	}{
+		{
+			name:         "empty url defaults to config endpoint",
+			wantEndpoint: "wss://example.com",
+		},
+		{
+			name:         "url variant a",
+			url:          "https://sumo-open-events.example.com",
+			wantEndpoint: "wss://sumo-opamp-events.example.com/v1/opamp",
+		},
+		{
+			name:         "url variant b",
+			url:          "https://sumo-open-collectors.example.com",
+			wantEndpoint: "wss://sumo-opamp-collectors.example.com/v1/opamp",
+		},
+		{
+			name:         "url variant c",
+			url:          "https://example.com",
+			wantEndpoint: "wss://example.com/v1/opamp",
+		},
```

Comment:
Can we add at least one test case for actual API urls this will convert?

Reply:
Sure, not a problem. :+1: